src/class/hash.rs

```diff
@@ -173,6 +250,12 @@ impl Hash {
     }
 }
 
+impl Clone for Hash {
+    fn clone(&self) -> Hash {
+        Hash { value: hash::dup(self.value()) }
```

Comment:
I don't know whether this is the way we want to deal with `*_dup` functions, I'm happy to discuss.